ledger/stateproofverificationtracker.go

```diff
@@ -0,0 +1,340 @@
+
+	if currentStateProofInterval == 0 {
+		return
+	}
+
+	if blk.Round()%currentStateProofInterval == 0 {
+		spt.appendCommitContext(&blk)
+	}
+
+	if delta.StateProofNext != 0 {
+		spt.appendDeleteContext(&blk, &delta)
+	}
+}
+
+func (spt *stateProofVerificationTracker) committedUpTo(round basics.Round) (minRound, lookback basics.Round) {
+	return round, 0
+}
+
+func (spt *stateProofVerificationTracker) produceCommittingTask(_ basics.Round, _ basics.Round, dcr *deferredCommitRange) *deferredCommitRange {
+	return dcr
+}
+
+func (spt *stateProofVerificationTracker) prepareCommit(dcc *deferredCommitContext) error {
+	spt.mu.RLock()
+	defer spt.mu.RUnlock()
+
+	lastContextToCommitIndex := spt.roundToLatestCommitContextIndex(dcc.newBase)
+	dcc.spVerification.CommitContext = make([]verificationCommitContext, lastContextToCommitIndex+1)
+	copy(dcc.spVerification.CommitContext, spt.trackedCommitContext[:lastContextToCommitIndex+1])
+
+	dcc.spVerification.LastDeleteIndex = spt.roundToLatestDeleteContextIndex(dcc.newBase)
+	if dcc.spVerification.LastDeleteIndex >= 0 {
+		dcc.spVerification.EarliestLastAttestedRound = spt.trackedDeleteContext[dcc.spVerification.LastDeleteIndex].stateProofNextRound
+	}
+
+	return nil
+}
+
+func (spt *stateProofVerificationTracker) commitRound(ctx context.Context, tx *sql.Tx, dcc *deferredCommitContext) (err error) {
+	err = insertStateProofVerificationContext(ctx, tx, dcc.spVerification.CommitContext)
+	if err != nil {
+		return err
+	}
+
+	if dcc.spVerification.LastDeleteIndex >= 0 {
+		err = store.DeleteOldStateProofVerificationContext(ctx, tx, dcc.spVerification.EarliestLastAttestedRound)
+	}
+
+	return err
+}
+
+func (spt *stateProofVerificationTracker) postCommit(_ context.Context, dcc *deferredCommitContext) {
+	spt.mu.Lock()
+	defer spt.mu.Unlock()
+
+	spt.trackedCommitContext = spt.trackedCommitContext[len(dcc.spVerification.CommitContext):]
+	spt.trackedDeleteContext = spt.trackedDeleteContext[dcc.spVerification.LastDeleteIndex+1:]
+}
+
+func (spt *stateProofVerificationTracker) postCommitUnlocked(context.Context, *deferredCommitContext) {
+
+}
+
+func (spt *stateProofVerificationTracker) handleUnorderedCommit(*deferredCommitContext) {
+
+}
+func (spt *stateProofVerificationTracker) close() {
+	if spt.dbQueries != nil {
+		spt.dbQueries.Close()
+		spt.dbQueries = nil
+	}
+}
+func (spt *stateProofVerificationTracker) LookupVerificationContext(stateProofLastAttestedRound basics.Round) (*ledgercore.StateProofVerificationContext, error) {
+	if lstlookup := spt.retrieveFromCache(stateProofLastAttestedRound); lstlookup != nil {
+		return lstlookup, nil
+	}
+
+	verificationContext, err := spt.lookUpVerificationContext(stateProofLastAttestedRound)
+	if err != nil {
+		return nil, err
+	}
+
+	spt.mu.Lock()
+	spt.lastLookedUpVerificationContext = *verificationContext
+	spt.mu.Unlock()
+
+	return verificationContext, nil
+}
+
+func (spt *stateProofVerificationTracker) retrieveFromCache(
```

Comment:
probably does not make sense to have it as a separate function - it trivially checks `spt.lastLookedUpVerificationContext`, and only used in `LookupVerificationContext` - consider inlining